### ISSUE TITLE
Setting output config by name after it has been set by identifier doesn't work on sway startup but does on reload

### ISSUE THREAD

**Issue by pedrocr** [swaywm/sway#5632]
### Please fill out the following:
- **Sway Version:** sway version v1.5-108b8e97 (Aug 19 2020, branch 'HEAD')

- **Debug Log:** https://gist.github.com/pedrocr/d618d47d76e7196882f65b014f6bbbac

- **Configuration File:** Default config plus some output configs like these:

```
output "Goldstar Company Ltd LG HDR 4K 0x0000333A" mode 3840x2160
output DP-1 mode 1920x1080@60hz
```

The identifier is for the monitor attached to the `DP-1` output.

- **Description:** Adjusting the output mode of the connection after the identifier doesn't work on sway startup. If I then do a reload of the config the correct mode is applied.

**Comment by WhyNotHugo** [comment]
https://github.com/swaywm/sway/blob/02b412a3d4e930237a1d16554af6e1e7d1855c89/sway/config/output.c#L698-L700

This comment indicates that during startup, the output don't exist -- which is true as per my testing; during initial config load, `WL-1` is not present, but it does show up during reloads.

The comment also indicates that during startup, the "new output event" should apply the config. I dug down through that, and found the cause of this issue: `get_output_config` has a fixed order of precedence, and `identifier` always gets priority (this is even clarified in a comment, so it's likely by design?).

There's then three ways to fix the issue at hand:

- The fix in a single commit from #5629. This one is a bit too complex, but works. If it sounds okay, I can open a PR with just that.
- Respecting the original order in `get_output_config`. This is likely less invasive, and can be done by just refactoring that single function (I'm assuming the items in the list match the original order).
- When reloading config, apply the same rule, and have the rule using an identifier always take precedence. This is probably the trickiest approach, but would match what seems to be a deliberate choice in `get_output_config`.

**Comment by emersion** [comment]
> I'm assuming the items in the list match the original order

I don't think this is the case? If you configure "DP-1", then configure "BenQ EL2870U F8K02862SL0" (same output), then configure "DP-1" again, I think the last configure will update the first item in the list?

**Comment by WhyNotHugo** [comment]
Ah, I hadn't though of that scenario.

I guess that means that respecting the order of entries (when one uses an identifier and another a name) in the config file would require a large refactor than expected, and, [as per this comment](https://github.com/swaywm/sway/blob/0cd8efe/sway/config/output.c#L641-L644), it seems that's not the intended behaviour by design anyway.

I guess this narrows down the options to:

1. The approach taken in this commit: https://github.com/swaywm/sway/commit/ac4f6ed583ae6913d9f23f7657b8aab752c39697. It works, but I fear it might add complexity.
2. When reloading config, have the rule using an identifier always take precedence. This is probably the trickiest approach, but would match what seems to be a deliberate choice mentioned above. But the gist of it is: when doing a reload merely update the `config->output_configs` objects, and _after_ having parsed the entire file, call `apply_output_config` for each output -- which end up being very similar to what happens during startup.

I'd also like to address this in a way that paves the road for #6410, and I think the second approach might make that a bit easier.

---

Just to be clear though, the current implementation means that if I've set up `output "Ancor Communications Inc..." scale 1.5` and then run `output * scale 1`, the former output is _not_ affected by the wildcard, since the per-name config always supersedes it.
